sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/aio/_schema_registry_avro_encoder_async.py

```diff
@@ -264,36 +310,78 @@ async def decode(
             try:
                 content = message["content"]
                 content_type = message["content_type"]
-            except (KeyError, TypeError):
-                SchemaDecodeError(
-                    f"""The content model {str(message)} is not a subtype of the MessageType protocol or type
-                        MessageContent.  If using an Azure SDK model class, please check the README.md
+            except (KeyError, TypeError) as exc:
+                raise TypeError(
+                    f"""The content model {str(message)} must be a subtype of the MessageType protocol or type
+                        MessageContent. If using an Azure SDK model class, please check the README.md
                         for the full list of supported Azure SDK models and their corresponding versions."""
-                ).raise_with_traceback()
+                ) from exc
 
-        # include in first preview for back compatibility
-        content, content_type = self._convert_preamble_format(content, content_type)
-
-        schema_id = content_type.split("+")[1]
+        try:
+            schema_id = content_type.split("+")[1]
+        except AttributeError:
+            raise InvalidContentError(
+                "Content type was not in the expected format of MIME type + schema ID."
+            )
```

Comment:
nit: I think sync and async share line 305 - 325, we could de-duplicate into one helper method.

feel free to change it in a separate/future PR